File: javascript/compresswords.js
function compressString(input) {
  let result = " ";
  let count = " ";

  for (let i = 1; i <= input.length; i++) {
    if (input[i] === input[i - 1]) {
      count++;
    } else {
      result += count + input[i - 1];
      count = 1;
    }
  }
  return result;
}
// const input = "aaaaabbbbbaaaccc";
// const output = compressString(input);
// console.log(output);
// jab output 5a5b3a3c chahiye ho

// final output

function compressBytotalCount(input) {
  let result = "";
  let seen = "";
  let counts = {};

  for (let char of input) {
    if (!seen.includes(char)) {
      seen += char;
    }
    counts[char] = (counts[char] || 0) + 1;
  }
  for (let char of seen) {
    result += counts[char] + char;
  }
  return result;
}

const input = "aaabbbbbaaabccc";
const output = compressBytotalCount(input);
console.log(output);

File: python/compresswords.py
def compressword(input_str):
    result = ""
    seen = ""
    counts = {}

    for char in input_str:
        if char not in seen:
            seen += char
        counts[char] = counts.get(char, 0) + 1
    for char in seen:
        result += str(counts[char]) + char
    return result


input_str = "aaaabbbaaabbccc"
output_str = compressword(input_str)
print(output_str)
